.github/scripts/upcoming_api_releases.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+set -eou pipefail
+
+
+URL="https://raw.githubusercontent.com/mongodb/openapi/dev/openapi/v2/versions.json"
+
+# Fetch the version.json file
+response=$(curl -s "${URL}")
+
+# Parse the dates from the JSON response using jq
+dates=$(echo "${response}" | jq -r '.[]')
+
+# Initialize an empty list to store dates within 3 weeks
+dates_within_3_weeks=()
+
+# Get the current date in seconds since epoch
+current_date=$(date +%s)
+
+# Determine if the system is macOS or Linux
+if [[ "$(uname)" == "Darwin" ]]; then
+    # macOS date command format
+    date_command="date -j -f %Y-%m-%d"
+else
+    # Linux date command format
+    date_command="date -d"
+fi
+
+# Iterate through each date
+for date in $dates; do
+    # Convert the date to seconds since epoch with explicit format
+    date_in_seconds=$($date_command "${date}" +%s 2>/dev/null)
+
+    # Calculate the difference in days between the date and the current date
+    diff_in_days=$(( (date_in_seconds - current_date) / (60 * 60 * 24) ))
+
+    # Check if the date is within 3 weeks (21 days)
+    if [[ "${diff_in_days}" -ge 0 && "${diff_in_days}" -le 21 ]]; then
+        # Add the date to the list if within 3 weeks
+        dates_within_3_weeks+=("${date}")
+    fi
+done
+
+if [[ ${#dates_within_3_weeks[@]} -gt 0 ]]; then
+    echo "API Versions that will be release in the next 3 weeks: ${dates_within_3_weeks[*]}"
```

Comment:
this will get quite noisy (every day, for 3 weeks in a row, we'll get this notification and tag @apix-2-on-call).
Is it too complicated to have this trigger conditions?
- isMonday and an API version is going to be released withing the next 3 weeks
- isWeekday and an API version is going to be released withing the 6 days (this is the release week)

Reply:
> isWeekday and an API version is going to be released withing the 6 days (this is the release week)

Isn't this still too noisy during the release week? I think that having the message only on Monday should be enough. WDYT?

Reply:
updated